.github/workflows/bicep-build.yml

```diff
@@ -0,0 +1,92 @@
+## deploy azDeploySecureSub.bicep
+
+name: Azure REDCAP Deployment
+
+on:
+    workflow_dispatch:
+
+    push:
+        branches:
+            - main
+
+    # pull_request:
+    #     branches:
+    #         - main
+
+permissions:
+    id-token: write
+    contents: read
+
+env:
+#   bicepFile: ./azDeploySecureSub.bicep
+    azCliVersion: 2.5.72
+            
+jobs:
+    # Validate the Bicep templates
+    validateTemplates:
+        runs-on: ubuntu-latest
+        
+        steps:
+            - uses: action/checkout@v4
+              name: Checkout
+
+            - uses: azure/login@v1
+              name: Azure Login
+              with:
+                client-id: ${{ secrets.AZURE_CLIENT_ID }}
+                tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+                subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+            - uses: azure/cli@v1
+              name: Azure CLI
+              with:
+                azcliversion: ${{ env.azCliVersion }}
+                inlineScript: |
+                    az deployment sub validate --location westeurope --template-file ./azDeploySecureSub.bicep
+    
+    # validate the resources before creation
+    validateResources:
```

Comment:
Do we really need different jobs, or are different steps sufficient? We could avoid having to do repetitive checkouts and logins.

Reply:
We will keep only two jobs: Validate and Deploy.